### PR TITLE
Tool API Routes

### PR DESCRIPTION
**Pull request by daniel-sm-yu**
## Notion ticket link

[API Routes (Tool)](https://www.notion.so/uwblueprintexecs/Task-Board-485f846ae00243599dddba2be9bcba82?p=744ad877c0264383a4c547204c8037d7)

## Implementation description
* created functions for each of the endpoints on [Notion](https://www.notion.so/uwblueprintexecs/API-Routes-eeb9e4403fba4676922868228329016b)
* dynamic routes in `[id].js`
* regular routes in `index.js`

## Steps to test
1. make requests to the endpoints

## What should reviewers focus on?
* 

## Checklist
- [ ] My PR name is descriptive and in imperative tense
- [ ] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [ ] I have run the appropriate linter(s)
- [ ] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: pages/api/tool/create.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import { Tool } from "database/models/tool";
```
Comment (nimwijetunga) [inline pages/api/tool/create.ts:1]: The usual naming convention for the post file is `/api/tool/post` (at least this is what is being used elsewhere in the codebase)

File: pages/api/tool/deleteOne.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+const deleteOne = async (
+    req: NextApiRequest,
+    res: NextApiResponse,
+): Promise<void> => {
+    const { id } = req.query;
+
+    await Tool.findByIdAndDelete(id)
+        .then((tool) => res.status(200).json(tool))
+        .catch((err) => res.status(500).send(err));
```
Comment (nimwijetunga) [inline pages/api/tool/deleteOne.ts:12]: will this throw an error if no tool for given id is found?

File: pages/api/tool/getOne.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+const getOne = async (
+    req: NextApiRequest,
+    res: NextApiResponse,
+): Promise<void> => {
+    const { id } = req.query;
+
+    await Tool.findById(id)
+        .then((tool) => res.status(200).json(tool))
+        .catch((err) => res.status(500).send(err));
```
Comment (nimwijetunga) [inline pages/api/tool/getOne.ts:12]: will this throw an error if no tool for the given id is found?

**Review by nimwijetunga — COMMENTED** [review]
few minor edits

File: pages/api/tool/deleteOne.ts (hunk 2 above)
Comment (daniel-sm-yu) [reply pages/api/tool/deleteOne.ts:12]: this won't throw an error if no tool has that id - should we throw an error/return 404?

File: pages/api/tool/getOne.ts (hunk 3 above)
Comment (daniel-sm-yu) [reply pages/api/tool/getOne.ts:12]: it won't - I'll change it to return 404 if no tools have the given id

**Review by nimwijetunga — APPROVED** [review]
lgtm

**Review by christineNg1028 — APPROVED** [review]

**Pull request merged**